### ISSUE TITLE
Incorrect force_unwrapping violation with array IUO

### ISSUE THREAD

**Issue by ldiqual** [realm/SwiftLint#1710]
On 0.20.1, the following code triggers a force_unwrapping violation:

```
var invites: [Invite]!
```

![image](https://user-images.githubusercontent.com/941453/28597737-7b23170a-7154-11e7-9c01-5bf63c553553.png)

I believe it is due to https://github.com/realm/SwiftLint/pull/1620

**Comment by marcelofabri** [comment]
Are you sure you're using 0.20.1? I couldn't reproduce it.

```
$ echo "var invites: [Invite]\!" | swiftlint lint --use-stdin --enable-all-rules 
Loading configuration from '.swiftlint.yml'
<nopath>:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
<nopath>:1:1: warning: Implicitly Unwrapped Optional Violation: Implicitly unwrapped optionals should be avoided when possible. (implicitly_unwrapped_optional)
<nopath>:1: warning: File Header Violation: Files should have consistent header comments. (file_header)
Done linting! Found 3 violations, 0 serious in 1 file.
```

**Comment by otaviolima** [comment]
It is reproducible with 0.21.1.

It is triggered when an implicitly unwrapped array variable is declared inside a function, closure, brackets, etc.

For instance:

```
$ echo "if cond { var invites: [Invite]! }" | swiftlint lint --use-stdin --enable-all-rules
<nopath>:1:32: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.
```
Or:

```
$ echo "abc { var invites: [Invite]! }" | swiftlint lint --use-stdin --enable-all-rules
<nopath>:1:28: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.
```

#1620 should have fixed this. But I was not aware of  SourceKitten limitation ( https://github.com/realm/SwiftLint/issues/136, https://github.com/jpsim/SourceKitten/issues/151) when reporting local variable declaration at the time I wrote the fix for subscripts. 

I've been trying to fix it for a while but no progress so far.

**Comment by marcelofabri** [comment]
But then, this is a duplicate of #136. There's not much we can do other than fixing the issue in SourceKit itself ((SR-5057)[https://bugs.swift.org/browse/SR-5057]).
